### PR TITLE
Replace usage of google.api.core with google.api_core

### PR DESCRIPTION
**Pull request by jonparrott**
This is the big one- this replaces any usage of `api.core` with `api_core` and makes `google-cloud-core` depend on `google-api-core`.

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -10,10 +10,8 @@
 core/*                    @dhermes
 test_utils/*              @dhermes
 
-# Jon Wayne Parrott is the author of the new google.api.core library.
-# NOTE: This trumps the previous core/* directive, since the last match is
-#       what matters.
-core/google/api/core/*    @jonparrott
+# Jon Wayne Parrott is the author of the new google.api_core library.
```
Comment (dhermes) [inline .github/CODEOWNERS]: Do we want to call it "new" in here? I assume this doc will exist long after it isn't new.

Comment (jonparrott) [reply .github/CODEOWNERS]: I expect this doc will cease to exist when we split the repo.

Comment (dhermes) [reply .github/CODEOWNERS]: DERP

**Comment by jonparrott** [comment]
Good call on updating get_target_packages, done.

**Pull request merged**